sklearn/feature_extraction/tests/test_text.py

```diff
@@ -328,6 +328,31 @@ def test_fit_countvectorizer_twice():
     assert_not_equal(X1.shape[1], X2.shape[1])
 
 
+def test_countvectorizer_custom_token_pattern():
+    corpus = [
+        'This is the first document.',
+        'This document is the second document.',
+        'And this is the third one.',
+        'Is this the first document?',
+    ]
+    # With no capturing group
+    token_pattern = r"[a-z]{2,}|(?:[0-9]{1,3})(?:st|nd|rd|th)?"
+    vectorizer = CountVectorizer(token_pattern=token_pattern)
+    vectorizer.fit_transform(corpus)
+    expected = ['and', 'document', 'first', 'is', 'one',
+                'second', 'the', 'third', 'this']
+    assert_equal(vectorizer.get_feature_names(), expected)
+    # With 2 capturing groups
+    token_pattern = r"([a-z]{2,})|([0-9]{1,3})(?:st|nd|rd|th)?"
+    try:
+        vectorizer.set_params(token_pattern=token_pattern)
+        vectorizer.fit_transform(corpus)
+        assert False, "we shouldn't get here"
+    except ValueError as e:
+        assert_in("more than 1 capturing group in token pattern",
```

Comment:
Usually we use assert_raise_message or assert_raises_regexp or `with pytest.raises`